src/renderer/wg_engine/tvgWgGeometry.h

```diff
@@ -208,15 +207,22 @@ struct WgVertexBuffer
                 pntIndex++;
             } else if (cmd == PathCommand::Close) {
                 close();
+                // close command creates the new object, started form the closing point
+                if ((vcount > 0) && (update_dist)) updateDistances();
+                if ((vcount > 0) && (onPolyline)) onPolyline(*this);
+                Point last_pt = last();
+                reset(tscale);
+                if (cmdIndex + 1 < rshape.path.cmds.count) append(last_pt);
```

Comment:
this is what I meant:
               ```
               close();
               if (cmdIndex == rshape...count || rshape...cmds[cmdIndex+1] != MoveTo) ccontinue;
               //rest of the code
                ```

```vcount > 0``` this can be done once